metaserver/src/main/scala/scala/meta/languageserver/ScalametaLanguageServer.scala

```diff
@@ -289,6 +300,15 @@ class ScalametaLanguageServer(
     sourceChangeSubscriber.onNext(input)
   }
 
+  override def executeCommand(request: WorkspaceExecuteCommandRequest) = Task {
+    import WorkspaceCommand._
+    WorkspaceCommand.withNameOption(request.params.command).map {
+      case ClearIndexCache =>
+        logger.info("Clearing the index cache")
+        ScalametaLanguageServer.clearCacheDirectory()
+    }.getOrElse(logger.error(s"Unknown command ${request.params.command}"))
```

Comment:
the server declares which messages it responds to, but that doesn't prevent the VSCode client to send arbitrary messages, so we log an error to help debugging typos and other mistakes.